cumulusci/core/dependencies/dependencies.py

```diff
@@ -0,0 +1,978 @@
+        # unpackaged/pre is always deployed unmanaged, no namespace manipulation.
+        deps.extend(
+            self._flatten_unpackaged(
+                repo, "unpackaged/pre", self.skip, managed=False, namespace=None
+            )
+        )
+
+        # Look for metadata under src (deployed if no namespace, or we're asked to do an unmanaged install)
+        if not managed:
+            contents = repo.directory_contents("src", ref=self.ref)
+            if contents:
+                deps.append(
+                    UnmanagedGitHubRefDependency(
+                        github=self.github,
+                        ref=self.ref,
+                        subfolder="src",
+                        unmanaged=self.unmanaged,
+                        namespace_inject=self.namespace_inject,
+                        namespace_strip=self.namespace_strip,
+                    )
+                )
+        else:
+            if self.managed_dependency is None:
+                raise DependencyResolutionError(
+                    f"Could not find latest release for {self}"
+                )
+
+            deps.append(self.managed_dependency)
+
+        # We always inject the project's namespace into unpackaged/post metadata if managed
+        deps.extend(
+            self._flatten_unpackaged(
+                repo,
+                "unpackaged/post",
+                self.skip,
+                managed=managed,
+                namespace=namespace,
+            )
+        )
+
+        return deps
+
+    @property
+    def name(self):
+        return f"Dependency: {self.github}"
+
+    @property
+    def description(self):
+        subfolder = f"/{self.subfolder}" if self.subfolder else ""
+        unmanaged = " (unmanaged)" if self.unmanaged else ""
+        loc = f" @{self.ref or self.tag}" if self.ref or self.tag else ""
+        return f"{self.name}{subfolder}{unmanaged}{loc}"
+
+
+class PackageNamespaceVersionDependency(StaticDependency):
+    """Static dependency on a package identified by namespace and version number."""
+
+    namespace: str
+    version: str
+    package_name: Optional[str]
+
+    @property
+    def package(self):
+        return self.package_name or self.namespace or "Unknown Package"
+
+    def install(
+        self,
+        context: BaseProjectConfig,
+        org: OrgConfig,
+        options: PackageInstallOptions = None,
+    ):
+        if not options:
+            options = PackageInstallOptions()
+
+        context.logger.info(f"Installing {self.package} version {self.version}")
+        install_package_by_namespace_version(
+            context,
+            org,
+            self.namespace,
+            self.version,
+            options,
+            retry_options=DEFAULT_PACKAGE_RETRY_OPTIONS,
+        )
+
+    @property
+    def name(self):
+        return f"Install {self.package} {self.version}"
+
+    @property
+    def description(self):
+        return self.name
```

Comment:
This can probably go in a base class as the default description.

Reply:
Done.